### PR TITLE
Promote plans for Free users when clicking ‘Share Schedule’ button

### PR DESCRIPTION
**Pull request by ezequielc**
## Description
Promote plans for Free users when clicking ‘Share Schedule’ button.
Create Unlock This Feature Modal that will be re-used for other features soon.
Do not show 'Share Schedule' button while adding a new Schedule (only shows after saved).

## Motivation and Context
Shared Schedules epic.

## How Has This Been Tested?
Locally and on stage-1.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

File: web/scripts/components/plans/ctr-unlock-this-feature-modal.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+angular.module('risevision.common.components.plans')
+  .controller('UnlockThisFeatureModalCtrl', [
+    '$scope', '$modalInstance', 'plansFactory',
+    function ($scope, $modalInstance, plansFactory) {
+
+      $scope.subscribe = function () {
+        plansFactory.showPlansModal();
```
Comment (alex-deaconu) [inline web/scripts/components/plans/ctr-unlock-this-feature-modal.js:9]: @ezequielc curious why not just use the modal promise result to showPlansModal and just use a regular confirm modal controller with a custom template and classes? Just a thought. Thanks.

Comment (ezequielc) [reply web/scripts/components/plans/ctr-unlock-this-feature-modal.js:9]: @alex-deaconu Good suggestion. I will try that. Thanks!

Comment (ezequielc) [reply web/scripts/components/plans/ctr-unlock-this-feature-modal.js]: @alex-deaconu I updated per suggestion and also improved some e2e tests that were failing. Thanks!

File: web/scripts/schedules/directives/dtv-schedule-fields.js [hunk 2]
```diff
@@ -43,11 +45,15 @@ angular.module('risevision.schedules.directives')
           };
 
           $scope.openSharedScheduleModal = function () {
-            $modal.open({
-              templateUrl: 'partials/schedules/shared-schedule-modal.html',
-              controller: 'SharedScheduleModalController',
-              size: 'md'
-            });
+            if (currentPlanFactory.isPlanActive() || currentPlanFactory.isCancelledActive()) {
```
Comment (alex-deaconu) [inline web/scripts/schedules/directives/dtv-schedule-fields.js:48]: @ezequielc does this include trials? Thanks.

Comment (ezequielc) [reply web/scripts/schedules/directives/dtv-schedule-fields.js:48]: @alex-deaconu Yes, trial is accounted for in `isPlanActive()`. Thanks

Comment (alex-deaconu) [reply web/scripts/schedules/directives/dtv-schedule-fields.js:48]: @ezequielc sounds good, thanks!

**Pull request merged**